### PR TITLE
add support for colorlight-i9

### PR DESCRIPTION
**Pull request by splinedrive**

### CODE REVIEW

**Comment by trabucayre** [comment]
LGTM
Just a small remark: constraints refers to [hdl/constraints](https://hdl.github.io/constraints/Data/Boards/index.html#colorlight-i5-v7-0). colorlight i5 is present but i9 no.
There is two possibilities: removing this line here, or adding the board in hdl/constraints (second option seems better) ;-)

**Comment by splinedrive** [comment]
> LGTM Just a small remark: constraints refers to [hdl/constraints](https://hdl.github.io/constraints/Data/Boards/index.html#colorlight-i5-v7-0). colorlight i5 is present but i9 no. There is two possibilities: removing this line here, or adding the board in hdl/constraints (second option seems better) ;-)

I extended hdl/constraints
https://github.com/hdl/constraints/pull/7

File: doc/boards.yml [hunk 1]
```diff
@@ -130,6 +130,14 @@
   Flash: OK
   Constraints: Colorlight-i5-v7.0
 
+- ID: colorlight_i9
```
Comment (trabucayre) [inline doc/boards.yml:133]: here you have an '_'

File: src/board.hpp [hunk 2]
```diff
@@ -122,6 +122,7 @@ static std::map <std::string, target_board_t> board_list = {
 	JTAG_BOARD("zybo_z7_20",      "xc7z020clg400",  "digilent", 0, 0, CABLE_DEFAULT),
 	JTAG_BOARD("colorlight",      "", "",           0, 0, CABLE_DEFAULT),
 	JTAG_BOARD("colorlight-i5",   "", "cmsisdap", 0, 0, CABLE_DEFAULT),
+	JTAG_BOARD("colorlight-i9",   "", "cmsisdap", 0, 0, CABLE_DEFAULT),
```
Comment (trabucayre) [inline src/board.hpp:125]: here you have '-'

**Comment by trabucayre** [comment]
> > LGTM Just a small remark: constraints refers to [hdl/constraints](https://hdl.github.io/constraints/Data/Boards/index.html#colorlight-i5-v7-0). colorlight i5 is present but i9 no. There is two possibilities: removing this line here, or adding the board in hdl/constraints (second option seems better) ;-)
> 
> I extended hdl/constraints [hdl/constraints#7](https://github.com/hdl/constraints/pull/7)

Great!

File: src/board.hpp (hunk 2 above)
Comment (splinedrive) [reply src/board.hpp:125]: Colorlight i5 has also '_' in id

Comment (trabucayre) [reply src/board.hpp:125]: True. There is a typo in i5

**Comment by trabucayre** [comment]
Applied to master.
Thank!

**Pull request merged**